meetings/2022-11-07.md

```diff
@@ -0,0 +1,107 @@
+
+
+### Issues
+
+https://github.com/solid/solid-oidc/issues
+
+[Reconsider Client Identifiers](https://github.com/solid/solid-oidc/issues/207)
+
+
+## Minutes
+
+### Open PRs
+
+* AC: Both Pavlik and I made PRs to authentication panel to clarify UTC. I'll close mine since Pavlik's has little more details.
+
+### Reconsider Client Identifiers #207
+
+Proposal: 
+
+> The Solid-OIDC specification should drop the current definition of Client Identifiers and associated JSON-LD appendix in favor of adopting the OpenID Connect Federation definition of an Entity Identifier and Automatic Registration
+
+* AC: With OAuth2 and OIDC it is hard to globally identify clients. In most cases client_id is local to the OP or AS. In Solid we need global client identifiers, and defined them in Solid-OIDC.
+* ...: Now there is the [OpenID Federation draft](https://openid.net/specs/openid-connect-federation-1_0.html
+ ). Among other things it defines mechanism to define clients globally. The question is if we can reuse it since it. The draft is at version 24. I have specific proposal which I would like to suggest. Let's take some questions first.
+* WT: I also went through most of the spec. I want to know what people see as possible obstacles.
+* ES: With migrating to OIDC Federation, what would the migration plan look like. The server would most likely need to support current version and next. How would we deal with the federation.
+* WT: I think it shouldn't be that hard. It sould be to easy to support current and next.
+* AC: To migration path, there are 3 parts to federation when it comes to client identifiers. The client / application interacts with various endponts using client_id query parameter with URL as value. This will not change.
+* ...: The two others are not as easy. On OP level, it will need to use the .well-known instead of just dereferencing the client_id. Check the signature chain etc. It is more complicated. On the plus side, since it is spec in OIDC family of specs. We can expect that OP providers will support it. The Solid specific provider should be able to make use of that. So in the short term there might be more work for OPs but in long term it should be more broadly available.
+* ...: Currently there is not requirement on signatures and trust chains for Client ID document. If we go with OIDC Federation, you can't just publish the document. It has to be signed and there must be trust chain.
+* WT: Do you know about existing implementations of OIDC Federation draft?
+* AC: I currently don't know any. I also don't know about any Solid Client ID outside of Solid.
+* WT: We implement it.
+* AC: I know, I just don't expect adoption outside of Solid-OIDC ecysystem.
+* ES: I understand Wouter talks about use.id
+* eP: For the trust chain, I haven't loooked deeply into that. It would rely heavily on the redirect_uri validation. The basic safety comes from that. What is the minimum needed for the trust/signature chain.
```

Comment:
```suggestion
* eP: I haven't looked deeply into the trust chain. It would rely heavily on `redirect_uri` validation. Basic safety comes from that. What is the minimum needed for the trust/signature chain?
```